components/selection/selection-header.js

```diff
@@ -0,0 +1,188 @@
+ */
+export class SelectionHeader extends SelectionObserverMixin(RtlMixin(LocalizeCoreElement(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether to render select-all and selection summary
+			 * @type {boolean}
+			 */
+			noSelection: { type: Boolean, attribute: 'no-selection' },
+			/**
+			 * Disables sticky positioning for the header
+			 * @type {boolean}
+			 */
+			noSticky: { type: Boolean, attribute: 'no-sticky' },
+			/**
+			 * Whether all pages can be selected
+			 * @type {boolean}
+			 */
+			selectAllPagesAllowed: { type: Boolean, attribute: 'select-all-pages-allowed' },
+			_hasActions: { state: true },
+			_scrolled: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				background-color: var(--d2l-selection-header-background-color, white);
+				display: block;
+				position: sticky;
+				top: 0;
+			}
+			:host([no-sticky]) {
+				background-color: transparent;
+				position: static;
+			}
+			.d2l-selection-header-shadow {
+				transition: box-shadow 200ms ease-out;
+			}
+			:host([_scrolled]) .d2l-selection-header-shadow {
+				background-color: var(--d2l-selection-header-background-color, white);
+				bottom: -4px;
+				box-shadow: 0 8px 12px -9px rgba(0, 0, 0, 0.3);
+				clip: rect(30px, auto, 200px, auto);
+				height: 40px;
+				position: absolute;
+				width: 100%;
+				z-index: -1;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-selection-header-container {
+				align-items: center;
+				display: flex;
+				margin-bottom: 6px;
+				margin-top: 6px;
+				min-height: 54px;
+			}
+			.d2l-selection-header-container-slim {
+				min-height: 36px;
+			}
+			d2l-selection-select-all {
+				flex: none;
+			}
+			d2l-selection-summary {
+				flex: none;
+				margin-left: 0.9rem;
+			}
+			:host([dir="rtl"]) d2l-selection-summary {
+				margin-left: 0;
+				margin-right: 0.9rem;
+			}
+			d2l-selection-select-all-pages {
+				flex: none;
+				margin-left: 0.45rem;
+			}
+			:host([dir="rtl"]) d2l-selection-select-all-pages {
+				margin-left: 0;
+				margin-right: 0.45rem;
+			}
+			.d2l-selection-header-actions {
+				--d2l-overflow-group-justify-content: flex-end;
+				flex: auto;
+				text-align: right;
+			}
+			:host([dir="rtl"]) .d2l-selection-header-actions {
+				text-align: left;
+			}
+			.d2l-sticky-edge {
+				left: 0;
+				margin-top: -1px;
+				position: absolute;
+				right: 0;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.noSelection = false;
+		this.noSticky = false;
+		this.selectAllPagesAllowed = false;
+		this._scrolled = false;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._stickyObserverDisconnect();
+	}
+
+	render() {
+		const classes = this._getSelectionHeaderContainerClasses();
+		return html`
+			<div class="d2l-sticky-edge"></div>
+			<div class="${classMap(classes)}">
+				${this.noSelection ? null : html`
+					<d2l-selection-select-all></d2l-selection-select-all>
```

Comment:
Do you have an eventual plan for the select-all? In the context of `d2l-list`, the select-all is contained within the `d2l-list-header` element, whereas in the context of `d2l-table-wrapper`, I think that consumers will be required to render it within their `<th>` element.

We might need to expose a slot where `d2l-list-header` can provide a `d2l-selection-select-all`. The alternative would be to change this from an inheritance to composition rendering approach where the sub-types have more rendering control.

Reply:
Yeah, I think we can just use the `noSelection` flag - the [design](https://rally1.rallydev.com/#/?detail=/userstory/656749634867&fdp=true) only shows "1-50 of 1,000 items" there, which I believe is just part of the "select all pages" logic. Still investigating that though.

The only issue there is that currently `noSelection` also omits the `d2l-selection-select-all-pages`.

Reply:
Ok, yeah, so I think we have some options. We can also introduce more granular properties, but at some point it just makes more sense let the user take more rendering control rather than baking it into this component. We can always tweak this in your next PR.

Reply:
Yeah I was thinking of tackling that in the next PR, especially since there's still uncertainty about what we actually want displayed there for tables.